server/index-node/src/resolver.rs

```diff
@@ -102,12 +93,62 @@ where
             .expect("Valid blockNumber required");
 
         let entity_changes = self
-            .subgraph_store
+            .store
+            .subgraph_store()
             .entity_changes_in_block(&subgraph_id, block_number)?;
 
         Ok(entity_changes_to_graphql(entity_changes))
     }
 
+    fn resolve_block_data(&self, field: &a::Field) -> Result<r::Value, QueryExecutionError> {
+        let network = field
+            .get_required::<String>("network")
+            .expect("Valid network required");
+
+        let block_hash = field
+            .get_required::<H256>("blockHash")
+            .expect("Valid blockHash required");
+
+        let chain_store = if let Some(cs) = self.store.block_store().chain_store(&network) {
+            cs
+        } else {
+            error!(
+                self.logger,
+                "Failed to fetch block data; nonexistant network";
+                "network" => network,
+                "block_hash" => format!("{}", block_hash),
+            );
+            return Ok(r::Value::Null);
+        };
+
+        let blocks_res = chain_store.blocks(&[block_hash]);
+        Ok(match blocks_res {
+            Ok(blocks) if blocks.is_empty() => {
+                error!(
+                    self.logger,
+                    "Failed to fetch block data; block not found";
+                    "network" => network,
+                    "block_hash" => format!("{}", block_hash),
+                );
+                r::Value::Null
+            }
+            Ok(mut blocks) => {
+                assert!(blocks.len() == 1, "Multiple blocks with the same hash");
```

Comment:
I wouldn't want this to crash a node, so I'd prefer if we just logged this with `error!`.

Reply:
I believe `"blocks_pkey" PRIMARY KEY, btree (hash)` guarantees us that the node will never fetch multiple blocks from a single hash

Reply:
So basically the `assert!` is there not to log an error but rather to provide error information in case we have an actual bug.

Reply:
Ok, true. That's fine then.